pkg/util/collections/includes_excludes_test.go

```diff
@@ -132,64 +133,121 @@ func TestValidateIncludesExcludes(t *testing.T) {
+			want:     []error{errors.New("excludes list cannot contain '*'")},
 		},
 		{
 			name:     "excludes cannot contain items in includes",
 			includes: []string{"foo", "bar"},
 			excludes: []string{"bar"},
-			expected: []error{errors.New("excludes list cannot contain an item in the includes list: bar")},
+			want:     []error{errors.New("excludes list cannot contain an item in the includes list: bar")},
 		},
 	}
 
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			res := ValidateIncludesExcludes(test.includes, test.excludes)
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			errs := ValidateIncludesExcludes(tc.includes, tc.excludes)
 
-			require.Equal(t, len(test.expected), len(res))
+			require.Equal(t, len(tc.want), len(errs))
 
-			for i := 0; i < len(test.expected); i++ {
-				assert.Equal(t, test.expected[i].Error(), res[i].Error())
+			for i := 0; i < len(tc.want); i++ {
+				assert.Equal(t, tc.want[i].Error(), errs[i].Error())
 			}
 		})
 	}
 }
 
 func TestIncludeExcludeString(t *testing.T) {
 	tests := []struct {
-		name             string
-		includes         []string
-		excludes         []string
-		expectedIncludes string
-		expectedExcludes string
+		name         string
+		includes     []string
+		excludes     []string
+		wantIncludes string
+		wantExcludes string
+	}{
+		{
+			name:         "unspecified includes/excludes should return '*'/'<none>'",
+			includes:     nil,
+			excludes:     nil,
+			wantIncludes: "*",
+			wantExcludes: "<none>",
+		},
+		{
+			name:         "specific resources should result in sorted joined string",
+			includes:     []string{"foo", "bar"},
+			excludes:     []string{"baz", "xyz"},
+			wantIncludes: "bar, foo",
+			wantExcludes: "baz, xyz",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			includesExcludes := NewIncludesExcludes().Includes(tc.includes...).Excludes(tc.excludes...)
+			assert.Equal(t, tc.wantIncludes, includesExcludes.IncludesString())
+			assert.Equal(t, tc.wantExcludes, includesExcludes.ExcludesString())
+		})
+	}
+}
+
+func TestValidateNamespaceIncludesExcludes(t *testing.T) {
+	tests := []struct {
+		name     string
+		includes []string
+		excludes []string
+		wantErr  bool
 	}{
 		{
-			name:             "unspecified includes/excludes should return '*'/'<none>'",
-			includes:         nil,
-			excludes:         nil,
-			expectedIncludes: "*",
-			expectedExcludes: "<none>",
+			name:     "empty slice doesn't return error",
+			includes: []string{},
+			wantErr:  false,
+		},
+		{
+			name:     "empty string is invalid",
+			includes: []string{""},
+			wantErr:  true,
 		},
 		{
-			name:             "specific resources should result in sorted joined string",
-			includes:         []string{"foo", "bar"},
-			excludes:         []string{"baz", "xyz"},
-			expectedIncludes: "bar, foo",
-			expectedExcludes: "baz, xyz",
+			name:     "asterisk by itself is valid",
```

Comment:
Can you add additional cases here to check the cases covered by `ValidateIncludesExcludes`? Such as checking that "*" in the excludes list is invalid and that using the same value in both includes and excludes is invalid.

I know this logic is already tested elsewhere, but I want to ensure that we fully verify the behaviour of this function.

Reply:
A few more tests have been added.